SwaMe/RTGrouper.cs

```diff
@@ -206,6 +207,14 @@ public RTMetrics DivideByRT(MzmlParser.Run run, int division, double rtDuration)
                 double ms2TicTotalTemp = 0;
                 foreach (MzmlParser.Scan scan in run.Ms2Scans)
                 {
+                    if (run.Ms1Scans.Count() < 1)
+                    {
+                        firstScanStartTime = Math.Min(firstScanStartTime, scan.ScanStartTime);
```

Comment:
There has to be a way of not making this evaluation in this loop? This might be done ~240,000 times

Reply:
We could take a lot of loop comparisons out by doing something like this:

firstScanStartTime = Math.Min(run.Ms1Scans.Min(x => x.ScanStartTime), run.Ms2Scans.Min(x => x.ScanStartTime));
lastScanStartTime = Math.Max(run.Ms1Scans.Max(x => x.ScanStartTime), run.Ms2Scans.Max(x => x.ScanStartTime));